GitCommands/Git/GitSshHelpers.cs

```diff
@@ -26,19 +26,26 @@ bool DoArgumentsRequireSsh()
             }
         }
 
-        /// <summary>Un-sets the git SSH command path.</summary>
-        public static void UnsetSsh()
-        {
-            Environment.SetEnvironmentVariable("GIT_SSH", "", EnvironmentVariableTarget.Process);
-        }
-
         /// <summary>Sets the git SSH command path.</summary>
         public static void SetSsh(string? path)
         {
-            if (!Strings.IsNullOrEmpty(path))
+            // Git will use the embedded OpenSSH ssh.exe if empty/unset
+            if (!string.IsNullOrEmpty(path) && !File.Exists(path))
             {
-                Environment.SetEnvironmentVariable("GIT_SSH", path, EnvironmentVariableTarget.Process);
+                path = "";
             }
+
+            if (!string.IsNullOrEmpty(path) && !File.Exists(path))
```

Comment:
Force pushing a backporting error (just unneeded, no negative effect as this is the path Git will use anyway).
```suggestion
            if (!string.IsNullOrEmpty(path))
```